libs/components/tabs/src/lib/modules/tabs/tabset-nav-button.component.ts

```diff
@@ -23,126 +20,174 @@ const buttonTypePrevious = 'previous';
   selector: 'sky-tabset-nav-button',
   templateUrl: './tabset-nav-button.component.html',
 })
-export class SkyTabsetNavButtonComponent implements AfterViewInit {
+export class SkyTabsetNavButtonComponent implements OnDestroy {
+  /**
+   * Specifies the tabset wizard component to associate with the nav button.
+   * @required
+   */
   @Input()
-  public tabset: SkyTabsetComponent;
+  public set tabset(value: SkyTabsetComponent) {
+    this.#_tabset = value;
 
-  @Input()
-  public buttonType: string;
+    if (this.#currentTabsetSub) {
+      this.#currentTabsetSub.unsubscribe();
+    }
+
+    if (value) {
+      if (this.#_buttonType === buttonTypeFinish) {
+        this.#_tabset.hasFinishButton = true;
+      }
+
+      this.#currentTabsetSub = this.#_tabset.activeChange
+        .pipe(distinctUntilChanged(), takeUntil(this.#ngUnsubscribe))
+        .subscribe((index: SkyTabIndex) => {
+          this.#activeSkyTabIndex = index;
+          this.#tabCount = this.#_tabset.tabs.length;
+          this.#updateTabToSelect();
+          this.#updateButtonVisibility();
+          this.#updateButtonProperties();
+        });
+    } else {
+      this.#logger.error(
```

Comment:
I wonder if we should actually throw an "Error" here instead of just logging and error and continuing execution. Thoughts?